### ISSUE TITLE
LDAP Authentication does't work

### ISSUE THREAD

**Issue by Jungle93** [allegro/ralph#3038]
As I was integrating the django-ldap auth .There were some problem:
I can fetch some ldap user information from remote server to add local user.but,It seems the password field does't encrypt with right algorithm.
with a django-ldap auth Demo project,there is the record :
===========================================================
CREATE TABLE "auth_user" ("id" integer NOT NULL PRIMARY KEY AUTOINCREMENT, "password" varchar(128) NOT NULL, "is_superuser" bool NOT NULL, "username" varchar(30) NOT NULL UNIQUE, "first_name" varchar(30) NOT NULL, "last_name" varchar(30) NOT NULL, "email" varchar(254) NOT NULL, "is_staff" bool NOT NULL, "is_active" bool NOT NULL, "date_joined" datetime NOT NULL, "last_login" datetime NULL)
table|django_session|django_session|14|CREATE TABLE "django_session" ("session_key" varchar(40) NOT NULL PRIMARY KEY, "session_data" text NOT NULL, "expire_date" datetime NOT NULL)
sqlite> select * from auth_user;

1|pbkdf2_sha256$20000$tUZRmXBhD3Z3$iymHnJXmKVEiQODOFjUayoau6ueQjoaIEAw1/RcWvZY=|1|jungle|||123456@wowowo.com|1|1|2017-04-26 04:58:16.889280|
2|!kE8xuPurGxadFffet4EzwXuboVZ3RHbbkdLF3jtD|0|michael|july|Sun|jungle93@git.com|0|1|2017-04-26 04:58:35.654008|2017-04-26 04:58:35.682058
sqlite>

============================================================
CREATE TABLE `accounts_ralphuser` (
  `id` int(11) NOT NULL AUTO_INCREMENT,
  `password` varchar(128) NOT NULL,
  `last_login` datetime DEFAULT NULL,
  `is_superuser` tinyint(1) NOT NULL,
  `username` varchar(30) NOT NULL,
  `first_name` varchar(30) NOT NULL,
  `last_name` varchar(30) NOT NULL,
  `email` varchar(254) NOT NULL,
  `is_staff` tinyint(1) NOT NULL,
  `is_active` tinyint(1) NOT NULL,
  `date_joined` datetime NOT NULL,
  `gender` int(10) unsigned NOT NULL,
  `country` int(10) unsigned NOT NULL,
  `city` varchar(30) NOT NULL,
  `company` varchar(64) NOT NULL,
  `employee_id` varchar(64) NOT NULL,
  `profit_center` varchar(1024) NOT NULL,
  `cost_center` varchar(1024) NOT NULL,
  `department` varchar(64) NOT NULL,
  `manager` varchar(1024) NOT NULL,
  `location` varchar(128) NOT NULL,
  `segment` varchar(256) NOT NULL,
  `team_id` int(11),
  PRIMARY KEY (`id`),
  UNIQUE KEY `username` (`username`),
  KEY `accounts_ralphuser_f6a7ca40` (`team_id`),
  CONSTRAINT `accounts_ralphuser_team_id_cfe4ebf580b8273_fk_accounts_team_id` FOREIGN KEY (`team_id`) REFERENCES `accounts_team` (`id`)
) ENGINE=InnoDB AUTO_INCREMENT=5 DEFAULT CHARSET=utf8;

with the ralph project ,there is the record:
'4', '!jennid4Vo3kae8CmawOScm6hJmSdL07CzdQal16h', NULL, '0', 'michael', 'july', 'Sun', 'jungle93@git.com', '0', '1', '2017-04-26 08:12:20', '2', '153', '', '', '', '', '', '', '', '', '', NULL
===============================================================================

the password field show above ,
pbkdf2_sha256$20000$tUZRmXBhD3Z3$iymHnJXmKVEiQODOFjUayoau6ueQjoaIEAw1/RcWvZY=

!=

!jennid4Vo3kae8CmawOScm6hJmSdL07CzdQal16h

Is that is the reason why i can't login into the ralph ?

**Comment by romcheg** [comment]
Closing a stale issue. If it is still relevant, please re-open it or start a topic at  Ralph forum: https://ralph.discourse.group

**Comment by jayooin** [comment]
I have the exact same problem. 
I did a synchronisation from the LDAP, Active Directory, and got all my users in the database. 
However, I could not log in.
They had the password starting with 
`!jennid4Vo3kae8CmawOScm6hJmSdL07CzdQal16h` as well.
I updated the passwords in the database, to something different, and the login worked.

The answer is that the Python module adds unusable passwords. There is no synchronisation on that side. 
As seen [here](https://django-auth-ldap.readthedocs.io/en/latest/users.html?highlight=password#user-objects) in its documentation: _Users created by LDAPBackend will have an unusable password set._
